<b>Numpy Project 1 Airport cab data Analysis <br>
    * Mean Speed of all rides <br>
    * Number of Rides taken in February <br>
    * Number of Rides where tip more than $50 <br>
    * Number of Rides where drop was JFK airport <br>

In [20]:
import numpy as np

In [21]:
taxi=np.genfromtxt('nyc_taxis.csv',delimiter=',',skip_header=True)

In [22]:
# Speed = Distance/Time
# we have taken here the 8 and 9 th column; 8th column 'Speed' and 9th column 'Trip_Length' in seconds.
# Here we need by hours. so, divide with 3600 
speed=taxi[:,7]/(taxi[:,8]/3600)

In [23]:
mean_speed=speed.mean()
print(mean_speed)

32.24258580925573


<b> Mean Speed of all rides : 32 miles per hour

In [24]:
# Calculate the number of Rides taken in the month of February.
# in our data set the value denoted by 2 on "Pick up month column"
taxi[:,1]==2

array([False, False, False, ..., False, False, False])

In [25]:
# We go for boolean and the row
rides_feb=taxi[taxi[:,1]==2,1]
print(rides_feb.shape[0])

13333


<b> Number of Rides taken in February is 13333

In [26]:
# Calculate the number of rides with a tip greater than $50
# Here in data set the "Tip Amunt column" presentin 3rd column from Last
Tip_amt=taxi[taxi[:,-3]>50,-3]
print(Tip_amt.shape[0])

16


<b> Number of Rides by people where tip more than $50 is 16

In [27]:
# Calculate the number of rides where drop was JFK airport
dropoff=taxi[taxi[:,6]==2,6]
print(dropoff.shape[0])

11832


<b> the number of rides where drop was JFK airport is 11832